awx/ui_next/src/screens/Job/JobOutput/HostEventModal.jsx

```diff
@@ -0,0 +1,218 @@
+    border-width: var(--pf-c-tabs__item--BorderWidth) 0
+      var(--pf-c-tabs__item--BorderWidth) 0;
+  }
+`;
+
+function HostEventModal({ handleClose, hostEvent, isOpen, i18n }) {
+  const [hostStatus, setHostStatus] = useState(null);
+  const [activeTabKey, setActiveTabKey] = useState(0);
+
+  useEffect(() => {
+    processEventStatus(hostEvent);
+  }, []);
+
+  const handleTabClick = (event, tabIndex) => {
+    setActiveTabKey(tabIndex);
+  };
+
+  function processEventStatus(event) {
+    let status = null;
+    if (event.event === 'runner_on_unreachable') {
+      status = 'unreachable';
+    }
+    // equiv to 'runner_on_error' && 'runner_on_failed'
+    if (event.failed) {
+      status = 'failed';
+    }
+    // catch the 'changed' case before 'ok', because both can be true
+    if (event.changed) {
+      status = 'changed';
+    }
+    if (
+      event.event === 'runner_on_ok' ||
+      event.event === 'runner_on_async_ok' ||
+      event.event === 'runner_item_on_ok'
+    ) {
+      status = 'ok';
+    }
+    if (event.event === 'runner_on_skipped') {
+      status = 'skipped';
+    }
+    setHostStatus(status);
+  }
+
+  function processStdOutValue() {
+    const { res } = hostEvent.event_data;
+    let stdOut;
+    if (taskAction === 'debug' && res.result && res.result.stdout) {
+      stdOut = processCodeMirrorValue(res.result.stdout);
+    } else if (
+      taskAction === 'yum' &&
+      res.results &&
+      Array.isArray(res.results)
+    ) {
+      stdOut = processCodeMirrorValue(res.results[0]);
+    } else {
+      stdOut = processCodeMirrorValue(res.stdout);
+    }
+    return stdOut;
+  }
+
+  const processCodeMirrorValue = value => {
+    let codeMirrorValue;
+    if (value === undefined) {
+      codeMirrorValue = false;
+    } else if (value === '') {
+      codeMirrorValue = ' ';
+    } else if (typeof value === 'string') {
+      codeMirrorValue = entities.encode(value);
+    } else {
+      codeMirrorValue = value;
+    }
+    return codeMirrorValue;
+  };
+
+  const taskAction = hostEvent.event_data.task_action;
+  const JSONObj = processCodeMirrorValue(hostEvent.event_data.res);
+  const StdErr = processCodeMirrorValue(hostEvent.event_data.res.stderr);
+  const StdOut = processStdOutValue();
+
+  return (
+    <Modal
+      isLarge
+      isOpen={isOpen}
+      onClose={handleClose}
+      title={i18n._(t`Host Details`)}
+      actions={[
+        <Button key="cancel" variant="secondary" onClick={handleClose}>
+          {i18n._(t`Close`)}
+        </Button>,
+      ]}
+    >
+      <Tabs activeKey={activeTabKey} onSelect={handleTabClick}>
+        <Tab eventKey={0} title={i18n._(t`Details`)}>
+          <DetailList style={{ alignItems: 'center' }} gutter="sm">
+            <Detail
+              label={i18n._(t`Host Name`)}
+              value={
+                <HostNameDetailValue>
+                  {hostStatus && <HostStatusIcon status={hostStatus} />}
```

Comment:
Here we could just use StatusIcon.

Reply:
Ah, thank you for making this suggestion! This consolidates and simplifies the status icon logic.